llvm/test/CodeGen/AArch64/neon-dotreduce.ll

```diff
@@ -2866,6 +4497,362 @@ entry:
   %x = add i32 %r1, %r2
   ret i32 %x
 }
+
+define i32 @test_usdot_v33i8(ptr nocapture readonly %a, ptr nocapture readonly %b, i32 %sum) {
```

Comment:
Do we need all these non-standard vector types in the tests?

Reply:
Probably not, but should the same rule be applied to duplicates of `dot`/`sdot`? Which test do you think would be enough to keep?

Reply:
There are quite a few of them! So long as there are tests for the usdot (including commuted patterns), then it should be OK. You can at least drop all the test that are not power-of-2 sizes - v5/v33/v24/v25.